src/helper/p2pool.rs

```diff
@@ -491,13 +491,15 @@ impl Helper {
 
         // 4. Loop as watchdog
         let mut first_loop = true;
+        let mut last_p2pool_request = Arc::new(Mutex::new(tokio::time::Instant::now()));
+        let mut last_status_request = Arc::new(Mutex::new(tokio::time::Instant::now()));
```

Comment:
any reason to use Arc<Mutex<_>> ? I think we can skip the overhead.

Reply:
No specific reason, I looked into the other files and this is how it was implemented so I went along with it.

Reply:
You can remove the Arc<Mutex<_>> . It is wrapped into it in [spawn_xvb_watchdog](https://github.com/Cyrix126/gupaxx/blob/d706dec3fcb861f34dae0ebbd9f1a995656b9ec8/src/helper/xvb/mod.rs#L151) method since it is shared accross a tokio spawn, which is not the case here.